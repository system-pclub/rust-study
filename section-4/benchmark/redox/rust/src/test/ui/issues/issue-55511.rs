#![warn(indirect_structural_match)]
use std::cell::Cell;
trait Foo<'a> {
    const C: Option<Cell<&'a u32>>;
}

impl<'a, T> Foo<'a> for T {
    const C: Option<Cell<&'a u32>> = None;
}

fn main() {
    let a = 22;
    let b = Some(Cell::new(&a));
    //~^ ERROR `a` does not live long enough [E0597]
    match b {
        <() as Foo<'static>>::C => { }
        //~^ WARN must be annotated with `#[derive(PartialEq, Eq)]`
        //~| WARN will become a hard error in a future release
        _ => { }
    }
}
