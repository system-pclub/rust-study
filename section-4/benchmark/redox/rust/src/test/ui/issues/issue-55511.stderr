warning: to use a constant of type `std::cell::Cell` in a pattern, `std::cell::Cell` must be annotated with `#[derive(PartialEq, Eq)]`
  --> $DIR/issue-55511.rs:16:9
   |
LL |         <() as Foo<'static>>::C => { }
   |         ^^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/issue-55511.rs:1:9
   |
LL | #![warn(indirect_structural_match)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #62411 <https://github.com/rust-lang/rust/issues/62411>

error[E0597]: `a` does not live long enough
  --> $DIR/issue-55511.rs:13:28
   |
LL |     let b = Some(Cell::new(&a));
   |                            ^^ borrowed value does not live long enough
...
LL |         <() as Foo<'static>>::C => { }
   |         ----------------------- type annotation requires that `a` is borrowed for `'static`
...
LL | }
   | - `a` dropped here while still borrowed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0597`.
